File: fastify_server/routes/healthz_vault_file_route.mjs
import { EnvStore } from "vault-file";

/**
 * Mount vault-file health check routes to the Fastify application.
 * This function is called by the server bootstrap process.
 * @param {import('fastify').FastifyInstance} server
 */
export async function mount(server) {
  server.get("/healthz/admin/vault-file/status", async (request, reply) => {
    const instance = EnvStore.getInstance();
    return {
      initialized: EnvStore.isInitialized(),
      totalVarsLoaded: instance["_totalVarsLoaded"] || 0,
    };
  });

  server.get("/healthz/admin/vault-file/json", async (request, reply) => {
    const instance = EnvStore.getInstance();
    const store = instance["store"] || {};
    return {
      initialized: EnvStore.isInitialized(),
      totalVarsLoaded: instance["_totalVarsLoaded"] || 0,
      store: { ...store },
    };
  });

  server.get("/healthz/admin/vault-file/keys", async (request, reply) => {
    const instance = EnvStore.getInstance();
    const store = instance["store"] || {};
    return {
      initialized: EnvStore.isInitialized(),
      totalVarsLoaded: instance["_totalVarsLoaded"] || 0,
      keys: Object.keys(store),
    };
  });
}

File: fastapi_server/routes/healthz_vault_file_route.py
from fastapi import FastAPI
from vault_file import EnvStore


def mount(app: FastAPI):
    """
    Mount vault-file health check routes to the FastAPI application.
    This function is called by the server bootstrap process.
    """

    @app.get("/healthz/admin/vault-file/status")
    async def vault_file_status():
        """Return the initialization status of the vault file EnvStore."""
        instance = EnvStore.get_instance()
        return {
            "initialized": EnvStore.is_initialized(),
            "total_vars_loaded": instance._total_vars_loaded,
        }

    @app.get("/healthz/admin/vault-file/json")
    async def vault_file_json():
        """Return the loaded vault file variables as JSON for debugging."""
        instance = EnvStore.get_instance()
        return {
            "initialized": EnvStore.is_initialized(),
            "total_vars_loaded": instance._total_vars_loaded,
            "store": dict(instance.store),
        }

    @app.get("/healthz/admin/vault-file/keys")
    async def vault_file_keys():
        """Return only the keys loaded from the vault file (no values)."""
        instance = EnvStore.get_instance()
        return {
            "initialized": EnvStore.is_initialized(),
            "total_vars_loaded": instance._total_vars_loaded,
            "keys": list(instance.store.keys()),
        }
